.gitignore

```diff
@@ -25,3 +25,52 @@ htmlcov
 .vscode
 .idea
 .pytest_cache/
+
+# For .ipynb examples
+docs/examples/advanced/functions_and_tuning_curves.py
+docs/examples/advanced/inhibitory_gating.py
+docs/examples/advanced/izhikevich.py
+docs/examples/advanced/matrix_multiplication.py
+docs/examples/advanced/nef_algorithm.py
+docs/examples/advanced/nef_summary.py
+docs/examples/advanced/processes.py
+docs/examples/basic/2d_representation.py
+docs/examples/basic/addition.py
+docs/examples/basic/combining.py
+docs/examples/basic/communication_channel.py
+docs/examples/basic/many_neurons.py
+docs/examples/basic/multiplication.py
+docs/examples/basic/single_neuron.py
+docs/examples/basic/squaring.py
+docs/examples/basic/two_neurons.py
+docs/examples/dynamics/controlled_integrator.py
+docs/examples/dynamics/controlled_integrator2.py
+docs/examples/dynamics/controlled_oscillator.py
+docs/examples/dynamics/integrator.py
+docs/examples/dynamics/lorenz_attractor.py
+docs/examples/dynamics/oscillator.py
+docs/examples/learning/learn_associations.py
+docs/examples/learning/learn_communication_channel.py
+docs/examples/learning/learn_product.py
+docs/examples/learning/learn_square.py
+docs/examples/learning/learn_unsupervised.py
+docs/examples/networks/basal_ganglia.py
+docs/examples/networks/ensemble_array.py
+docs/examples/networks/integrator_network.py
+docs/examples/quirks/config.py
+docs/examples/spa/associative_memory.py
+docs/examples/spa/convolution.py
+docs/examples/spa/question.py
+docs/examples/spa/question_control.py
+docs/examples/spa/question_memory.py
+docs/examples/spa/spa_parser.py
+docs/examples/spa/spa_sequence.py
+docs/examples/spa/spa_sequence_routed.py
+docs/examples/usage/config.py
+docs/examples/usage/delay_node.py
+docs/examples/usage/exceptions.py
+docs/examples/usage/network_design.py
+docs/examples/usage/network_design_advanced.py
+docs/examples/usage/rectified_linear.py
+docs/examples/usage/strings.py
+docs/examples/usage/tuning_curves.py
```

Comment:
There shouldn't be any that we need to keep I don't think

Reply:
I had wanted to keep it open in case we get any `.py` examples, but that's a good point that we do that `rm` anyway, so we should just ignore them all as long as we have that line in there.